### PR TITLE
rose task-run: allow -O (key) syntax

### LINKED ISSUE #1694: rose task-run: allow specified optional configuration files option to be missing
We allow this within an app:
`opts=ketchup (mayonnaise)`

It would be useful to do something like this in the suite.
Demo suite.rc:
```
[cylc]
[scheduling]
   [[dependencies]]
      graph = bar => foo => baz
[runtime]
   [[bar]]
        script = rose task-run --app-key=bar -O item1 -O (item2)
   [[baz]]
        script = rose task-run --app-key=baz --opt-conf-key=item3 --opt-conf-key=(item4)  
    [[foo]]
        script = rose task-run --app-key=foo
        [[[environment]]]
            ROSE_APP_OPT_CONF_KEYS = item5 (item6)
```
where item2, item4, item6 optional configurations are only used if they exist in the opt/ directory.

### PR DESCRIPTION
**Pull request by matthewrmshin**
Allow missing optional configuration keys via CLI option and
environment variable.

Close #1694.

### CODE REVIEW

**Comment by matthewrmshin** [comment]
@kaday @arjclark please review.

**Comment by kaday** [comment]
I am happy that it works and the tests run as expected.

Please add/update documentation, including the quotes around the options as previously they were not required.
For example
```
 command scripting = rose task-run --app-key=bar -O 'bye' -O '(hello)'
```
Works, but
```
 command scripting = rose task-run --app-key=bar -O bye -O (hello)
```
Does not.

**Comment by matthewrmshin** [comment]
No, `rose task-run --app-key=bar -O bye -O (hello)` will never have worked, because when the shell sees an open bracket, it expects to have to open a sub-shell.

**Comment by matthewrmshin** [comment]
New syntax documented.

**Comment by kaday** [comment]
Looks good to me, over to 2nd review

File: bin/rose-suite-run [hunk 1]
```diff
@@ -88,7 +88,9 @@
 #         Do not validate (suite engine configuration) in strict mode.
 #     --opt-conf-key=KEY, -O KEY
```
Comment (arjclark) [inline bin/rose-suite-run]: Looks like this line needs updating too, should be:

--opt-conf-key=KEY, -O KEY, --opt-conf-key='(KEY)', -O '(KEY)'

File: doc/rose-command.html [hunk 2]
```diff
@@ -2229,7 +2232,9 @@ <h2 id="rose-suite-run">rose suite-run</h2>
       <dt><kbd>--opt-conf-key=KEY, -O KEY</kbd></dt>
```
Comment (arjclark) [inline doc/rose-command.html]: Looks like needs updating as above.

**Comment by arjclark** [comment]
Bear with me. Having major test-battery problems on this branch.

**Comment by arjclark** [comment]
Am getting errors across the test battery due to this:
```
  File "/net/home/h06/aclark/rose.git/lib/python/rose/config_tree.py", line 116, in load
    for opt_key in opt_keys:
TypeError: 'NoneType' object is not iterable
```

**Comment by matthewrmshin** [comment]
Should now be better.

**Comment by arjclark** [comment]
Yup. Looks good to me.

**Pull request merged**